### DISCUSSION TITLE
Migration from third-party link field plugins to new Link field

### DISCUSSION THREAD

**Discussion by petehjfd** [craftcms/cms#15993] [Ideas]
Hi

The new native [Link field](https://craftcms.com/docs/5.x/reference/field-types/link.html) is great.  Until now, we've used various third-party plugins that have come and gone over the years to provide this functionality:

- [LinkIt](https://plugins.craftcms.com/linkit)
- [Typed Link Field](https://plugins.craftcms.com/typedlinkfield)
- [Hyper](https://plugins.craftcms.com/hyper)

We have a lot of sites that use one of the above, and would love to migrate them over to the new native Link field.  However, manually performing that migration without losing content will be very time consuming.

Would Pixel & Tonic ever consider creating migraton scripts to handle this?

Thanks!

**Comment by ammannbe** [comment]
I've found this simple and semi-automated migration for [Typed Link Field](https://plugins.craftcms.com/typedlinkfield): https://github.com/sebastian-lenz/craft-linkfield/issues/286#issuecomment-2569359736

@brandonkelly maybe still a consideration worth, to integrate it into Craft CMS directly? Regarding https://github.com/craftcms/cms/pull/15251#issuecomment-2251476665
I know it's hard to manage migrations for multiple plugins. But I think at least [Typed Link Field](https://plugins.craftcms.com/typedlinkfield) would be a consideration worth since it's widely used and was abandoned (but still works in Craft CMS 5.x).

**Comment by MangoMarcus** [comment]
Is there a solution for Hyper

**Comment by alexanderloewe** [comment]
Just write your own migration script.
